nixos/modules/system/boot/systemd/tmpfiles.nix

```diff
@@ -254,5 +276,62 @@ in
       "R! /nix/var/nix/gcroots/tmp           -    -    -    - -"
       "R! /nix/var/nix/temproots             -    -    -    - -"
     ];
+
+    boot.initrd.systemd = {
+      additionalUpstreamUnits = [
+        "systemd-tmpfiles-setup-dev.service"
+        "systemd-tmpfiles-setup.service"
+      ];
+      # make sure all the /dev nodes are set up
+      services.systemd-tmpfiles-setup-dev.wantedBy = ["sysinit.target"];
```

Comment:
I *think* this is actually handled by `sysinit.target.wants`: https://github.com/NixOS/nixpkgs/blob/66ea67f394126fce803fb9ee03f9097637655eb0/nixos/modules/system/boot/systemd/initrd.nix#L76-L78

Reply:
Hmm, I can't see how. Am I missing something?
https://github.com/systemd/systemd/blob/v256.4/units/sysinit.target